pkg/client/client.go

```diff
@@ -267,6 +268,20 @@ func (c *Client) NamespacesToMonitor() ([]string, error) {
 	return namespaceNames, nil
 }
 
+func (c *Client) IsHighlyAvaliable() (bool, error) {
+	infra, err := c.GetInfrastructure("cluster")
+	if err != nil {
+		return true, errors.Wrap(err, "could not get Infrastructure 'cluster' instance")
+	}
+	klog.V(6).Infof("control plane topology is: %s", infra.Status.ControlPlaneTopology)
+
+	if infra.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode {
```

Comment:
it might be more future-proof to just return `infra.Status.ControlPlaneTopology` [and/or `infra.Status.InfrastructureTopology`][1], instead of collapsing to a boolean here.  That would allow you to `switch` on the value in consuming code, and you could expand those switches later if the API grows more topology modes and monitoring has to do something more than default them to HA.

[1]: https://github.com/openshift/enhancements/pull/671#discussion_r584803866